### DISCUSSION TITLE
What was the motivation behind the creation of the Erlang Term Format package?

### DISCUSSION THREAD

**Discussion by KelvinHu** [discord-jda/JDA#2143] [General]
I was pleasantly surprised to see the implementation of a decoder and encoder for the ETF whilst browsing the Javadoc. So as in the title, what's the motivation behind it?

**Comment by DV8FromTheWorld** [comment]
Discord itself supports sending and receiving data over the Websocket via JSON _or_ ETF due their realtime websocket API being built on Erlang. The ETF encoding is actually also used by the Discord client itself when communicating over Websocket, and in general is a faster more compressed representation of the data when compared to JSON.

As such, we implemented the encoding so that developers can choose for themselves which to use when dealing with Discord.

This setting is accessible via [JDABuilder#setGatewayEncoding(GatewayEncoding)](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/JDABuilder.html#setGatewayEncoding(net.dv8tion.jda.api.GatewayEncoding))

**Reply by KelvinHu** [reply]
Thanks for the answer :)

Is there any downside to using ETF for the gateway over JSON?

As an aside, where did you find out about the the Discord client using ETF?

**Reply by Tais993** [reply]
Documentation:

[Developer docs, gateway ETF](https://discord.com/developers/docs/topics/gateway#etfjson)
